### PR TITLE
Add Poetry

### PR DESCRIPTION
**Pull request by sondrelg**
This PR adds Poetry for package management.

Changes:
- Replaced setup.py with Poetry and pyproject.toml configuration
- I added `isort` for import management in the pre-commit config (say the word and we can revert this)
- I removed the `__version__` declaration in `__init__.py` since I think this relies on the setup.py setup
- I added a new github action workflow for publishing pytest-split to pypi - you might not want to use this, but I thought I'd include it just in case. If you do want to try it, you will need to add the two secrets referenced in the workflow 🙂

Are you familiar with Poetry already @jerry-git? If not I would recommend just reading their [docs](https://python-poetry.org/docs/) - they're pretty short and very good 👍 

The tl;dr is

- `poetry install` to install all dependencies 
- `poetry update` to update dependencies
- `poetry build` to build the package (though that's usually not something I would do locally ever)
- `poetry add <package>` to add a dependency
- `poetry add -D <package>` to add a dev dependency

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,58 @@
+[tool.poetry]
+name = "pytest-split"
+version = "0.2.1"
```
Comment (jerry-git) [inline pyproject.toml:3]: How is this updated? Manually?

**Comment by jerry-git** [comment]
There's btw still the travis setup which publishes to PyPI on tags https://github.com/jerry-git/pytest-split/blob/beed24dda22387c56e4f584583f4f2eb346b9c03/.travis.yml#L10-L16

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: publish package on pypi
+
+on:
+  release:
+    types: [published, edited]
```
Comment (jerry-git) [inline .github/workflows/publish.yml:5]: Why on edited?

File: pyproject.toml (hunk 1 above)
Comment (sondrelg) [reply pyproject.toml:3]: Yeah so when I update a package I would manually indent the version here and publish a release in the github repo 🙂

File: .github/workflows/publish.yml (hunk 2 above)
Comment (sondrelg) [reply .github/workflows/publish.yml:5]: No reason really - I can remove it. Think I added it to a project when the initial release failed because of bad credentials in the repo.

Pypi won't allow you to re-republish a package though, so you could in theory run the workflow 100x and it would only publish successfully once.

**Comment by sondrelg** [comment]
> There's btw still the travis setup which publishes to PyPI on tags
> 
> https://github.com/jerry-git/pytest-split/blob/beed24dda22387c56e4f584583f4f2eb346b9c03/.travis.yml#L10-L16

Should I just delete the `travis.yml` file?

File: pyproject.toml (hunk 1 above)
Comment (jerry-git) [reply pyproject.toml:3]: I wonder if there's some way to get the version from git tags with poetry 🤔  But yeah, I can look into that later.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (jerry-git) [reply .github/workflows/publish.yml:5]: I assume `edited` will be triggered also with some simple typo fix? Let's have only `published` here 🙂

**Comment by jerry-git** [comment]
> Should I just delete the travis.yml file?

Yeah let's burn it 🔥

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -30,6 +30,10 @@ repos:
             'flake8-print',
             'flake8-type-checking',
         ]
+  - repo: https://github.com/pycqa/isort
```
Comment (jerry-git) [inline .pre-commit-config.yaml:33]: 👍

File: pyproject.toml (hunk 1 above)
Comment (sondrelg) [reply pyproject.toml:3]: https://github.com/tiangolo/poetry-version-plugin

Comment (sondrelg) [reply pyproject.toml:3]: If you want to add this in a follow-up PR you'd just need to change the test and publish action to this as well, since it requires >= 1.2.0a1

```yaml
      - name: Install poetry
        uses: snok/install-poetry@v1.1.6
        with:
          version: 1.2.0a1
          virtualenvs-in-project: true

```

**Comment by jerry-git** [comment]
Seems to work nicely, 0.2.1 was published using the GHA workflow added here 👍

**Comment by sondrelg** [comment]
Nice 🚀
